azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -196,13 +195,13 @@ def _publish_events_in_queue(self, event_data):
         logger.info("checking event queue")
         while True:
             try:
-                (event_to_send, callback) = self._event_queue.get_nowait()
+                (message_to_send, callback) = self._event_queue.get_nowait()
             except queue.Empty:
                 logger.info("done checking queue")
                 return
             logger.info("retrieved event from queue. publishing.")
-            topic = self._get_telemetry_topic()
-            mid = self._mqtt_provider.publish(topic, event_to_send)
+            encoded_topic = self._encode_properties(message_to_send, self.topic)
+            mid = self._mqtt_provider.publish(encoded_topic, message_to_send.data)
```

Comment:
question for the masses: node, at least, doesn't require a message object. to wrap the data.  What if the user just passes down a dictionary or a string?  Do we fail and make then wrap it in a Message, or do we send it as-is?

Reply:
we should probably wrap in a message and then send. 

---
In reply to: [240783586](https://github.com/Azure/azure-iot-sdk-python-preview/pull/34#discussion_r240783586) [](ancestors = 240783586)